index.js

```diff
@@ -356,9 +363,17 @@ class ClientServiceBase {
             if (err) {
               reject(err);
             } else {
-              const jsonResponse = response.toObject();
-              jsonResponse.result = JSON.parse(jsonResponse.result);
-              resolve(jsonResponse);
+              const resultInString = response.getResult();
+              const resultInObject = (resultInString)
+                ? JSON.parse(resultInString)
+                : {};
+
+              const contractExecutionResult = new ContractExecutionResult(
+                  resultInObject,
+                  response.getProofsList().map((proof) => proof.toObject()),
```

Comment:
This `proof` is still grpc generated `AssetProof`, right ? 
Can we convert it into our defined class like we do in Java ?
I'm thinking using grpc generated classes is not very user friendly.
What do you think ?

Reply:
Yes. It is gRPC's AssetProof.
The reason that I didn't encapsulate it is the gRPC's AssetProof class has almost the same functions of AssetProof class in Java SDK.

AssetProof in Java SDK
```
public String getId();
public int getAge();
public byte[] getHash();
public String getNonce();
public byte[] getSignature();
```

gRPC's AssetProof class
```
// @return {string}
proto.rpc.AssetProof.prototype.getAssetId();

// @return {number}
proto.rpc.AssetProof.prototype.getAge();

// @return {!(string|Uint8Array)}
proto.rpc.AssetProof.prototype.getHash();

// @return {string}
proto.rpc.AssetProof.prototype.getNonce();

// @return {!(string|Uint8Array)}
proto.rpc.AssetProof.prototype.getSignature();
```
But, there is `toString()` implemented in AssetProof in Java SDK.
So, I think we can provide the same thing in JavaScript SDK.

Reply:
I see. But as far as I can see,  there are other grpc specific methods that users won't even use, right ?
The following is the result of grep in `scalar_pb.js`.

```
proto.rpc.AssetProof.prototype.toObject = function(opt_includeInstance) {
proto.rpc.AssetProof.toObject = function(includeInstance, msg) {
proto.rpc.AssetProof.deserializeBinary = function(bytes) {
proto.rpc.AssetProof.deserializeBinaryFromReader = function(msg, reader) {
proto.rpc.AssetProof.prototype.serializeBinary = function() {
proto.rpc.AssetProof.serializeBinaryToWriter = function(message, writer) {
proto.rpc.AssetProof.prototype.getAssetId = function() {
proto.rpc.AssetProof.prototype.setAssetId = function(value) {
proto.rpc.AssetProof.prototype.getAge = function() {
proto.rpc.AssetProof.prototype.setAge = function(value) {
proto.rpc.AssetProof.prototype.getNonce = function() {
proto.rpc.AssetProof.prototype.setNonce = function(value) {
proto.rpc.AssetProof.prototype.getHash = function() {
proto.rpc.AssetProof.prototype.getHash_asB64 = function() {
proto.rpc.AssetProof.prototype.getHash_asU8 = function() {
proto.rpc.AssetProof.prototype.setHash = function(value) {
proto.rpc.AssetProof.prototype.getSignature = function() {
proto.rpc.AssetProof.prototype.getSignature_asB64 = function() {
proto.rpc.AssetProof.prototype.getSignature_asU8 = function() {
proto.rpc.AssetProof.prototype.setSignature = function(value) {
```

Reply:
Yes. Indeed.
To have another class is better in this case.
I create AssetProof class.
Please kindly review :)

Reply:
Maybe a bit more re-factory.

Reply:
OK !